src/lwc/psElevateTokenHandler/psElevateTokenHandler.js

```diff
@@ -90,23 +98,30 @@ class psElevateTokenHandler {
     * Rejects any messages from an unknown origin.
     */
     registerPostMessageListener(component) {
-        let self = this;
-
+        const self = this;
         window.onmessage = async function (event) {
-
-            if (self._visualforceOriginUrls) {
-                self._visualforceOrigin = self._visualforceOriginUrls.find(
-                    origin => event.origin === origin.value
-                ).value;
-            }
-
-            if (self._visualforceOrigin) {
+              if (self.shouldHandleMessage(event)) {
                 const message = JSON.parse(event.data);
                 component.handleMessage(message);
             }
         }
     }
 
+    shouldHandleMessage (event) {
+        return this.isExpectedVisualForceOrigin(event) &&
+            validateJSONString(event.data);
+    }
+
+    isExpectedVisualForceOrigin (event) {
+        this._visualforceOrigin = this._visualforceOriginUrls.find(
+            origin => event.origin === origin.value
+        );
+        this._visualforceOrigin =
+            this._visualforceOrigin !== undefined ?
```

Comment:
Let's have conditional operator either all on one line or split, for example:
`condition ? exprIfTrue : exprIfFalse;` or
```
condition 
    ? exprIfTrue 
    : exprIfFalse;
```

Reply:
Updated!